Makefile.work

```diff
@@ -185,7 +187,18 @@ SONIC_BUILD_INSTRUCTION :=  make \
                            EXTRA_JESSIE_TARGETS=$(EXTRA_JESSIE_TARGETS) \
                            $(SONIC_OVERRIDE_BUILD_VARS)
 
-.PHONY: sonic-slave-build sonic-slave-bash init reset
+PDESUBMODULES = src/sonic-linux-kernel \
+                src/sonic-utilities \
+                src/sonic-platform-common \
+                src/sonic-swss-common \
+                src/sonic-py-swsssdk \
+                src/redis-dump-load \
+                src/sonic-sairedis \
+                src/sonic-platform-daemons \
+                src/sonic-platform-pde \
+                src/sonic-ztp
+
+.PHONY: sonic-slave-build sonic-slave-bash init initpde reset
```

Comment:
get rid of initpde

Reply:
We need to be able to build a minimal subset of SONiC with the PDE docker enabled. We can either handle this as part of make init (or in this case make initpde), or we would have to modify slave.mk to ignore NON-PDE required build targets.  In this case, we think its easier to maintain a list of essential submodules for the PDE and make it part of the "make initpde" step for doing a build.  Can we leave this change as it is or do yo have an alternative suggestion? 

We automate the PDE build internally by doing the following.

make initpde
make configure PLATFORM=broadcom
make target/sonic-broadcom-pde.bin

Reply:
@lguohan, @jleveque  - can you please address these responses from Bill (following comments also)? We have come up with a way of doing a minimal build for PDE (reduced build time etc), but it seems that you didn't like this. Do you have an alternative in mind? Thanks.

Reply:
> @lguohan, @jleveque - can you please address these responses from Bill (following comments also)? We have come up with a way of doing a minimal build for PDE (reduced build time etc), but it seems that you didn't like this. Do you have an alternative in mind? Thanks.

@lguohan, @jleveque - I never heard back. Should I setup a call to discuss?

Reply:
We need to run this "make initpde" to differentiate the Full SONIC image build. Actually, ODM will not run this step, since the step is done by BRCM internally to generate the light weight PDE repo for ODM. Another reason for this is to protect the BRCM SONIC changes with just platform related common change exposing to ODM when working with ODM on ODM HWQ projects.

For PDE, we also have another option which is to run PDE as a docker in Full SONIC image. In the FULL sonic image runtime, we can launch PDE by pde.sh script. The script will disable most dockers in SONIC image and just reserve the necessary dockers with a lightweight environment for ODM platform testing. By default, the PDE docker will not be triggered in Full SONIC image.

Geans

make initpde

make configure PLATFORM=broadcom

make target/sonic-broadcom-pde.bin

Reply:
@lguohan, @jleveque - no movement on this topic - please offer some morning slots and I will try to setup a call - thx.